cluster-autoscaler/simulator/predicatechecker/schedulerbased_test.go

```diff
@@ -36,52 +39,122 @@ func TestCheckPredicate(t *testing.T) {
 
 	n1000 := BuildTestNode("n1000", 1000, 2000000)
 	SetNodeReadyState(n1000, true, time.Time{})
+	n1000Unschedulable := BuildTestNode("n1000", 1000, 2000000)
+	SetNodeReadyState(n1000Unschedulable, true, time.Time{})
+
+	defaultPredicateChecker, err := NewTestPredicateChecker(nil)
+	assert.NoError(t, err)
+
+	// temp dir
+	tmpDir, err := os.MkdirTemp("", "scheduler-configs")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	customConfigFile := filepath.Join(tmpDir, "custom_config.yaml")
+	if err := os.WriteFile(customConfigFile, []byte(`
+apiVersion: kubescheduler.config.k8s.io/v1
+kind: KubeSchedulerConfiguration
+profiles:
+- pluginConfig:
+  plugins:
+    multiPoint:
+      disabled:
+      - name: NodeResourcesFit
+        weight: 1
+  schedulerName: custom-scheduler`), os.FileMode(0600)); err != nil {
+		t.Fatal(err)
+	}
+
+	customConfig, err := scheduler.GetSchedulerConfig(customConfigFile)
+	assert.NoError(t, err)
+	customPredicateChecker, err := NewTestPredicateChecker(customConfig)
+	assert.NoError(t, err)
 
 	tests := []struct {
-		name          string
-		node          *apiv1.Node
-		scheduledPods []*apiv1.Pod
-		testPod       *apiv1.Pod
-		expectError   bool
+		name             string
+		node             *apiv1.Node
+		scheduledPods    []*apiv1.Pod
+		testPod          *apiv1.Pod
+		predicateChecker PredicateChecker
+		expectError      bool
 	}{
+		// default predicate checker test cases
+		{
+			name:             "default - other pod - insuficient cpu",
+			node:             n1000,
+			scheduledPods:    []*apiv1.Pod{p450},
+			testPod:          p600,
+			expectError:      true,
+			predicateChecker: defaultPredicateChecker,
+		},
+		{
+			name:             "default - other pod - ok",
+			node:             n1000,
+			scheduledPods:    []*apiv1.Pod{p450},
+			testPod:          p500,
+			expectError:      false,
+			predicateChecker: defaultPredicateChecker,
+		},
+		{
+			name:             "default - empty - insuficient cpu",
+			node:             n1000,
+			scheduledPods:    []*apiv1.Pod{},
+			testPod:          p8000,
+			expectError:      true,
+			predicateChecker: defaultPredicateChecker,
+		},
 		{
-			name:          "other pod - insuficient cpu",
-			node:          n1000,
-			scheduledPods: []*apiv1.Pod{p450},
-			testPod:       p600,
-			expectError:   true,
+			name:             "default - empty - ok",
+			node:             n1000,
+			scheduledPods:    []*apiv1.Pod{},
+			testPod:          p600,
+			expectError:      false,
+			predicateChecker: defaultPredicateChecker,
 		},
+		// custom predicate checker test cases
 		{
-			name:          "other pod - ok",
-			node:          n1000,
-			scheduledPods: []*apiv1.Pod{p450},
-			testPod:       p500,
-			expectError:   false,
+			name:             "custom - other pod - ok",
```

Comment:
Test cases for custom predicate checker are a copy of the default predicate checker except we don't expect an error for any of the test cases.